Plugin.fs

```diff
@@ -9,39 +9,51 @@ open Dalamud.Game.ClientState
 open Dalamud.IoC
 open DalamudPluginProjectTemplateFSharp.Attributes
 
-type Plugin([<RequiredVersion("1.0")>] pluginInterface : DalamudPluginInterface,
-            [<RequiredVersion("1.0")>] commands : CommandManager,
-            [<RequiredVersion("1.0")>] chat : ChatGui,
-            [<RequiredVersion("1.0")>] clientState : ClientState) as this =
-    let mutable commandManager : PluginCommandManager<Plugin> = null
-    let mutable config = Configuration()
-    let ui = PluginUI()
+type Plugin
+    (
+        [<RequiredVersion("1.0")>] pluginInterface: DalamudPluginInterface,
+        [<RequiredVersion("1.0")>] commands: CommandManager,
+        [<RequiredVersion("1.0")>] chat: ChatGui,
+        [<RequiredVersion("1.0")>] clientState: ClientState
+    ) as this =
+    let config =
+        match pluginInterface.GetPluginConfig() with
+        | null -> Configuration()
+        | config -> downcast config
 
-    // Initialization
-    let loadedConfig = pluginInterface.GetPluginConfig()
-    do if (loadedConfig <> null) then
-        config <- downcast loadedConfig
     do config.Initialize(pluginInterface)
-    do pluginInterface.UiBuilder.add_Draw(fun() -> ui.Draw())
-    do commandManager <- new PluginCommandManager<Plugin>(this, commands)
+
+    /// Implement your plugin UI in the PluginUI class.
+    let ui = PluginUI()
+    /// Delegate instance for ui.Draw; reference saved for Dispose().
+    let drawHandler = Action(ui.Draw)
+    do pluginInterface.UiBuilder.add_Draw drawHandler
```

Comment:
I haven't been able to actually check if this is the correct way to handle delegates. For some reason F# is not translating the C# `event`s as documented in https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/events#handling-events, their documentation doesn't explain how to deal with `Remove` anyway, and the rest of the page only talks about exposing F#-native Events to C#.